### ISSUE TITLE
Enum serialisation issue with V7

### ISSUE THREAD

**Issue by morganleroi** [algolia/algoliasearch-client-csharp#861]
There is another breaking change about JSON serialization we have seen later: the `System.Text.Json` serialization is set in the Algolia client to convert enums to string, while previously, they were converted to integers.

Since the Algolia client registers a global converter for doing this, it seems the only ways to adapt is either to set a custom converter on all enum properties of serialized models, or type the enum properties as `int` and adapt the code fetching them.

_Originally posted by @fredericDelaporte in https://github.com/algolia/algoliasearch-client-csharp/issues/852#issuecomment-2322177297_

**Comment by morganleroi** [comment]
@fredericDelaporte could you point me out what `enum` do you use ? In the `v6` code base, there was only 3 real enum. All other properties that are now enum in `v7` were `string`.

**Comment by fredericDelaporte** [comment]
In our indexes data, we have properties typed as enums. That is our own business models enums. They were converted in the indexes as ints. Now, they are converted as strings.

**Comment by morganleroi** [comment]
Hello @fredericDelaporte  ! 

You are right, the default behaviour has changed and all enum are now saved as string in the index. 

But If you get the saved object using the client, then the enum is correctly deserialized as a classic enum. (You can get the Value and the corresponding `int`). 

Did I missed something ? 

``` csharp
using Algolia.Search.Clients;
using Algolia.Search.Models.Search;

var searchClient = new SearchClient("W7H0CJAYM5", "021c8c87dfeafab6309804336d6829de");

var saveOperation = await searchClient.SaveObjectAsync("MyIndex", new TestObj { Value = TestEnum.Value2 });
await searchClient.WaitForTaskAsync("MyIndex", saveOperation.TaskID);

var result = await searchClient.GetObjectsAsync<TestObj>(new GetObjectsParams
{ Requests =
  [new(saveOperation.ObjectID, "MyIndex")]
});

Console.WriteLine(result.Results.First().Value); // Output: Value2
Console.WriteLine((int)result.Results.First().Value ); // Output: 1

public class TestObj
{
  public TestEnum Value { get; set; }
}

public enum TestEnum
{
  Value1,
  Value2
}
```

**Comment by fredericDelaporte** [comment]
Additional consequences: we build the indexes with a .Net app but search them through JavaScript. So, JavaScript code expecting integers can not work. We have put converters on all our enum and enums array properties to get them serialized as int again.

(Sure enough, string values are more descriptive than integers, but they are also more susceptible to breaking changes due to renaming, and they cause the data to be heavier. Anyway, my point is just to notify a breaking change which should be included in the upgrade guide.)

**Comment by morganleroi** [comment]
Hello @fredericDelaporte 

Sorry for the late response. You are right, it's not an expected behaviour. We will publish a bug fix where the enum are persisted as `int`. We will also create a dedicated documentation about the enum behaviour + the Serialization changes.

**Comment by morganleroi** [comment]
@fredericDelaporte 

A new minor version has been released (`7.4.0`). This version fixes the enum issue. 

Please note that your enum will be saved at `int` in this version.

**Comment by fredericDelaporte** [comment]
Thanks.
